sqldb/migrations.go

```diff
@@ -381,6 +385,29 @@ func ApplyMigrations(ctx context.Context, db *BaseDB,
 			"(dirty=%v) as base version", currentVersion, dirty)
 	}
 
+	// Due to an a migration issue in v0.19.0-rc1 we may be at version 2 and
+	// have a dirty schema due to failing migration 3. If this is indeed the
+	// case, we need to reset the dirty flag to be able to apply the fixed
+	// migration.
+	if version == 2 {
+		schemaVersion, dirty, err := migrator.GetSchemaVersion()
+		if err != nil {
+			return err
+		}
+
+		if schemaVersion == 3 && dirty {
+			log.Warnf("Schema version %d is dirty. This is "+
+				"likely a consequence of a failed migration "+
+				"in v0.19.0-rc1. Attempting to recover by "+
+				"resetting the dirty flag", schemaVersion)
+
+			err = migrator.SetSchemaVersion(3, false)
```

Comment:
I think it'd be safe to move to version 4 here.

Reply:
I think it does not matter right now because you made migration 3 idempotent ?

Reply:
For someone who's just upgrading to 0.19 rc1 from 0.18.2 or lower applying migration 3 makes sense, however anyone ave 0.18.3 (https://github.com/lightningnetwork/lnd/pull/8855) should not re-apply that as it'd change their invoice expiries. I think it'd be safe to jump over it.

Reply:
leading on from my previous comment: but is it safe to assume here that migration 4 has actually already run in the past?